File: backend/src/main/java/com/github/bytestrick/tabula/controller/dto/table/TableContentDTO.java
package com.github.bytestrick.tabula.controller.dto.table;

import jakarta.validation.constraints.NotNull;

import java.util.List;
import java.util.UUID;

/**
 * Data Transfer Object (DTO) representing a table structure,
 * including its header (columns) and content (rows).
 *
 * @param id       the unique identifier (UUID) of the table; must not be {@code null}
 * @param header   list of {@link ColumnDTO} defining each column of the header's table; must not be {@code null}
 * @param content  list of {@link RowDTO} representing each row of the table; must not be {@code null}
 */
public record TableContentDTO(
        @NotNull UUID id,
        @NotNull List<ColumnDTO> header,
        @NotNull List<RowDTO> content
) {}

File: frontend/src/app/model/dto/table/table-content-dto.ts
import {ColumnDTO} from './column-dto';
import {RowDTO} from './row-dto';

/**
 * Represents a full table fetched from the backend, including its columns and rows.
 *
 * @property id
 *   UUID of the table.
 * @property header
 *   Array of {@link ColumnDTO} describing each header column.
 * @property content
 *   Array of {@link RowDTO} containing the table rows and their cell values.
 */
export interface TableContentDTO {
  id: string,
  header: ColumnDTO[],
  content: RowDTO[]
}
